pages/api/v1/bounty/bountyDashboard/getAllBounties.ts

```diff
@@ -0,0 +1,242 @@
+				.get();
+
+			bountiesDocs?.docs?.map((doc: any) => {
+				if (doc.exists) {
+					const data = doc.data();
+					bountiesIndexes.push(Number(data?.id));
+				}
+			});
+
+			totalBounties = (await postsByTypeRef(network, ProposalType.BOUNTIES).where('tags', 'array-contains-any', categories).count().get()).data().count;
+		}
+
+		const variables: any = {
+			limit: 10,
+			offset: BOUNTIES_LISTING_LIMIT * (page - 1)
+		};
+		if (statuses.length) {
+			variables.status_in = statuses;
+		}
+		if (bountiesIndexes?.length) {
+			variables.index_in = bountiesIndexes;
+		}
+		const subsquidBountiesRes = await fetchSubsquid({
+			network,
+			query: GET_ALL_BOUNTIES,
+			variables: variables
+		});
+
+		if (!subsquidBountiesRes?.data?.bounties?.length) throw apiErrorWithStatusCode('No bounty data found', 400);
+
+		const subsquidBountiesData = subsquidBountiesRes?.data?.bounties || [];
+		totalBounties = totalBounties > 0 ? totalBounties : subsquidBountiesRes?.data?.totalBounties?.totalCount || 0;
+
+		const subsquidbountiesIndexes = subsquidBountiesData.map((bounty: { index: number }) => bounty?.index);
+
+		bountiesDocs = bountiesDocs ? bountiesDocs : await postsByTypeRef(network, ProposalType.BOUNTIES).where('id', 'in', subsquidbountiesIndexes).get();
+
+		const bountiesPromises = subsquidBountiesData.map(async (subsquidBounty: ISubsquidBounty) => {
+			const subsquidChildBountiesRes = await fetchSubsquid({
+				network,
+				query: GET_ALL_CHILD_BOUNTIES_BY_PARENT_INDEX,
+				variables: {
+					parentBountyIndex_eq: subsquidBounty?.index
+				}
+			});
+
+			const subsquidChildBountyData = subsquidChildBountiesRes?.data?.proposals || [];
+			const totalChildBountiesCount = subsquidChildBountiesRes?.data?.proposalsConnection?.totalCount || 0;
+
+			let claimedAmount = ZERO_BN;
+
+			subsquidChildBountyData.map((childBounty: { status: string; reward: string }) => {
+				const amount = new BN(childBounty?.reward || 0);
+
+				if ([bountyStatus.CLAIMED, bountyStatus.AWARDED].includes(childBounty.status)) {
+					claimedAmount = claimedAmount.add(amount);
+				}
+			});
+
+			const payload: IBounty = {
+				categories: [],
+				claimedAmount: claimedAmount.toString(),
+				createdAt: subsquidBounty?.createdAt,
+				curator: subsquidBounty?.curator,
+				index: subsquidBounty.index,
+				payee: subsquidBounty?.payee,
+				proposer: subsquidBounty?.proposer,
+				reward: subsquidBounty?.reward,
+				status: subsquidBounty?.status,
+				title: '',
+				totalChildBountiesCount: totalChildBountiesCount || 0
+			};
+
+			bountiesDocs?.docs?.map((bountyDoc) => {
+				if (bountyDoc?.exists) {
+					const bountyData = bountyDoc?.data();
+					if (bountyData?.id == subsquidBounty.index) {
+						payload.title = bountyData?.title || '';
+						payload.categories = bountyData?.tags || [];
+					}
+				}
+			});
+
+			if (!payload?.title) {
+				const res = await getSubSquareContentAndTitle(ProposalType.BOUNTIES, network, subsquidBounty?.index);
```

Comment:
please add a dataSource: 'subsquare' or 'polkassembly' item in the object for where the context is from